scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/spel/internal/propertyAccessors.scala

```diff
@@ -51,34 +54,63 @@ object propertyAccessors {
     This one is a bit tricky. We extend ReflectivePropertyAccessor, as it's the only sensible way to make it compilable,
     however it's not so easy to extend and in interpreted mode we skip original implementation
    */
-  object NoParamMethodPropertyAccessor extends ReflectivePropertyAccessor with ReadOnly with Caching {
+  class NoParamMethodPropertyAccessor(set: ClassDefinitionSet)
+      extends ReflectivePropertyAccessor
+      with ReadOnly
+      with Caching {
+    private val methodInvoker = new ExtensionsAwareMethodInvoker(set)
+    private val emptyArray    = Array[AnyRef]()
 
     override def findGetterForProperty(propertyName: String, clazz: Class[_], mustBeStatic: Boolean): Method = {
       findMethodFromClass(propertyName, clazz).orNull
     }
 
     override protected def reallyFindMethod(name: String, target: Class[_]): Option[Method] = {
-      target.getMethods.find(m =>
-        !ClassUtils.isPrimitiveOrWrapper(target) && m.getParameterCount == 0 && m.getName == name
-      )
+      ExtensionAwareMethodsDiscovery
+        .discover(target)
+        .find(m => !ClassUtils.isPrimitiveOrWrapper(target) && m.getParameterCount == 0 && m.getName == name)
     }
 
     override protected def invokeMethod(
         propertyName: String,
         method: Method,
         target: Any,
         context: EvaluationContext
-    ): AnyRef = {
-      method.invoke(target)
-    }
+    ): Any =
+      methodInvoker.invoke(method)(target, emptyArray)
 
     override def getSpecificTargetClasses: Array[Class[_]] = null
+
+    override def createOptimalAccessor(context: EvaluationContext, target: Any, name: String): PropertyAccessor =
+      super.createOptimalAccessor(context, target, name) match {
+        case o: OptimalPropertyAccessor => new NuOptimalAccessor(o)
```

Comment:
What's going on here? Won't it itroduce a performance downgrade? Should it be used in every situation? We need to add a verbose comment here :)